e2e/test/context.go

```diff
@@ -84,6 +86,15 @@ func (c *Context) Context() context.Context {
 	return c.context
 }
 
+// WithDeadline returns a derived context with a deadline. Call cancel to release resources associated with the context
+// as soon as the operation running in the context complete.
+func (c Context) WithDeadline(d time.Time) (*Context, context.CancelFunc) {
```

Comment:
We can simplify the code by using WithTimeout so callers do not need to compute the deadline. context.WithTimeout compute the deadline internally.